### PR TITLE
Update k8s version to v1.22.3 in all docs

### LINKED ISSUE #268: Update BYOH Dockerfile to be consistent with v1.22.3
The current [BYOHDockerfile](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/test/e2e/BYOHDockerFile) which we use as BYOHosts in local-dev setup makes use of `k8s v1.22.0`. With the new [v0.1.0-alpha.2](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/releases/tag/v0.1.0-alpha.2) release, installer support is present only for `Ubuntu 20.04` and `k8s v1.22.3`. We have updated the guides to reflect the same.
This leaves the docker hosts experience to be out of sync with what we recommend in the guide.
An additional info about `skip-installation` flag needs to be added when using docker hosts or VMs that already have k8s installed.

### PR DESCRIPTION
**Pull request by anusha94**
**What this PR does / why we need it**:
- update byoh docker host image to v1.22.3
- update getting started guide and local-dev guide
- update default k8s version to v1.22.3 in e2e tests

Fixes #268

### CODE REVIEW

File: test/e2e/config/provider.yaml [hunk 1]
```diff
@@ -79,7 +79,7 @@ providers:
 variables:
   # default variables for the e2e test; those values could be overridden via env variables, thus
   # allowing the same e2e config file to be re-used in different prow jobs e.g. each one with a K8s version permutation
-  KUBERNETES_VERSION: "v1.22.0"
+  KUBERNETES_VERSION: "v1.22.3"
   ETCD_VERSION_UPGRADE_TO: "3.4.9-0"
   COREDNS_VERSION_UPGRADE_TO: "1.7.0"
   KUBERNETES_VERSION_UPGRADE_TO: "v1.22.0"
```
Comment (NilanjanDaw) [inline test/e2e/config/provider.yaml:85]: I think we need to update the `KUBERNETES_VERSION_UPGRADE_TO` variable also.

Comment (anusha94) [reply test/e2e/config/provider.yaml:85]: we don't have any upgarde tests. That is an unused variable. So left it as is.

**Review by dharmjit — APPROVED** [review]
LGTM

**Comment by NilanjanDaw** [comment]
LGTM

**Review by NilanjanDaw — APPROVED** [review]

**Pull request merged**